.config/eslint.mjs

```diff
@@ -0,0 +1,573 @@
+    'sort-keys': OFF, // TODO
+    'space-before-blocks': [ON, 'always'],
+    'space-before-function-paren': [
+      OFF, // Conflicts w/ Prettier.
+      {
+        anonymous: 'never',
+        named: 'never',
+        asyncArrow: 'always',
+      },
+    ],
+    'space-in-parens': [ON, 'never'],
+    'space-infix-ops': ON,
+    'space-unary-ops': ON,
+    'spaced-comment': [
+      ON,
+      'always',
+      {
+        block: { balanced: false },
+        exceptions: ['-', '**'],
+      },
+    ],
+    strict: [ON, 'global'],
+    'symbol-description': ON,
+    'template-curly-spacing': ON,
+    'unicode-bom': ON,
+    'use-isnan': ON,
+    'valid-typeof': ON,
+
+    // -------------------------------
+    // TypeScript ESLint Plugin Rules
+    // -------------------------------
+
+    '@typescript-eslint/array-type': [ON, {
+      default: 'generic',
+      readonly: 'generic',
+    }],
+    '@typescript-eslint/camelcase': OFF,
+    '@typescript-eslint/consistent-type-definitions': [ON, 'interface'],
+    '@typescript-eslint/explicit-function-return-type': OFF,
+    /** > This rule is dumb and only works with TypeScript syntax.
+     * — @DerekNonGeneric */
+    "@typescript-eslint/explicit-module-boundary-types": OFF,
+    '@typescript-eslint/no-explicit-any': ON,
+    '@typescript-eslint/no-non-null-assertion': OFF,
+    '@typescript-eslint/no-throw-literal': OFF,
+    '@typescript-eslint/no-unsafe-assignment': OFF,
+    '@typescript-eslint/no-unsafe-member-access': OFF,
+    '@typescript-eslint/no-unsafe-return': OFF,
+    '@typescript-eslint/no-use-before-define': OFF,
+    '@typescript-eslint/no-var-requires': OFF,
+    '@typescript-eslint/prefer-as-const': ON,
+    '@typescript-eslint/prefer-nullish-coalescing': OFF,
+    '@typescript-eslint/prefer-optional-chain': OFF,
+    '@typescript-eslint/restrict-template-expressions': OFF,
+    /**
+     * > This rule is an abomination and should never have been accepted into
+     * > eslint core, and should never be used by anyone. async function is
+     * > hugely useful by itself to statically ensure a promise is returned, and
+     * > thrown exceptions reject it. await is never needed. — @ljharb */
+    '@typescript-eslint/require-await': OFF,
+    '@typescript-eslint/triple-slash-reference': ON,
+    '@typescript-eslint/type-annotation-spacing': ON,
+    '@typescript-eslint/unbound-method': OFF,
+    '@typescript-eslint/unified-signatures': ON,
+
+    // ----------------------
+    // Prettier Plugin Rules
+    // ----------------------
+
+    'prettier/prettier': [ON, prettierOptions],
+
+    // --------------------
+    // No Null Plugin Rule
+    // --------------------
+
+    'no-null/no-null': ON,
+
+    // ------------------
+    // Node Plugin Rules
+    // ------------------
+
+    'node/file-extension-in-import': [OFF, 'always', { '.js': 'never' }], // TODO
+    'node/no-unsupported-features/es-syntax': [
+      ON,
+      {
+        version: pjsonObj.engines.node,
```

Comment:
this isn't necessary to specify, by default it reads from "engines" in package.json per [the docs](https://github.com/mysticatea/eslint-plugin-node/blob/HEAD/docs/rules/no-unsupported-features/es-syntax.md#configured-nodejs-version-range)

Reply:
TIL

Reply:
I'm going to leave it like this in case we decide to use `devEngines` in the future.

Reply:
I'm not aware that that's a thing, but also for something not published, what purpose would the distinction serve?

Reply:
For something not published, all `devDependencies` would use `devEngines`, but you're probably right that the distinction is pointless, so nevermind. :)